bugtool/cmd/configuration.go

```diff
@@ -0,0 +1,247 @@
+		"top -b -n 1",
+		"uptime",
+		"dmesg",
+		"bpftool map show",
+		"bpftool prog show",
+		// Versions
+		"docker version",
+		// Docker and Kubernetes logs from systemd
+		"journalctl -u cilium*",
+		"journalctl -u kubelet",
+		// iptables
+		"iptables-save",
+		"iptables -S",
+		"ip6tables -S",
+		"iptables -L -v",
+	}
+
+	// Commands that require variables and more configuration are added
+	// separately below
+	commands = append(commands, catCommands()...)
+	commands = append(commands, ethoolCommands()...)
+	commands = append(commands, copyConfigCommands(confDir, k8sPods)...)
+	commands = append(commands, copyCiliumInfoCommands(ciliumDir, k8sPods)...)
+	commands = append(commands, k8sCommands(commands, k8sPods)...)
+	return commands
+}
+
+func save(c *BugtoolConfiguration, path string) error {
+	f, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("Failed to open file %s for writing: %s\n", path, err)
+	}
+	defer f.Close()
+
+	data, err := json.MarshalIndent(c, "", "\t")
+	if err != nil {
+		return fmt.Errorf("Cannot marshal config %s", err)
+	}
+	err = ioutil.WriteFile(path, data, 0644)
+	if err != nil {
+		return fmt.Errorf("Cannot write config %s", err)
+	}
+	return nil
+}
+
+func loadConfigFile(path string) (*BugtoolConfiguration, error) {
+	var content []byte
+	var err error
+	logrus.WithField(logfields.Path, path).Debug("Loading file")
+
+	content, err = ioutil.ReadFile(path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var c BugtoolConfiguration
+	err = json.Unmarshal(content, &c)
+	return &c, err
+}
+
+func catCommands() []string {
+	files := []string{
+		"/proc/sys/net/core/bpf_jit_enable",
+		"/proc/kallsyms",
+		"/etc/resolv.conf",
+		"/var/log/upstart/docker.log",
+		"/var/log/docker.log",
+		"/var/log/daemon.log",
+		"/var/log/messages",
+	}
+	// Only print the files that do exist to reduce number of errors in
+	// archive
+	commands := []string{}
+	for _, f := range files {
+		if _, err := os.Stat(f); os.IsNotExist(err) {
+			continue
+		}
+		commands = append(commands, fmt.Sprintf("cat %s", f))
+	}
+	// TODO: handle K8s case as well.
+	return commands
+}
+
+func copyConfigCommands(confDir string, k8sPods []string) []string {
+	commands := []string{}
+	type Location struct {
+		Src string
+		Dst string
+	}
+
+	locations := []Location{
+		{"/proc/config", fmt.Sprintf("%s/kernel-config", confDir)},
+		{"/proc/config.gz", fmt.Sprintf("%s/kernel-config.gz", confDir)},
+	}
+
+	if len(k8sPods) == 0 {
+		kernel, _ := execCommand("uname", "-r")
+		kernel = strings.TrimSpace(kernel)
+		l := Location{fmt.Sprintf("/boot/config-%s", kernel),
+			fmt.Sprintf("%s/kernel-config-%s", confDir, kernel)}
+		locations = append(locations, l)
+
+		for _, location := range locations {
+			if _, err := os.Stat(location.Src); os.IsNotExist(err) {
+				continue
+			}
+			commands = append(commands, fmt.Sprintf("cp %s %s", location.Src, location.Dst))
+		}
+	} else {
+		for _, pod := range k8sPods {
+			prompt := podPrefix(pod, "uname -r")
```

Comment:
I don't understand what the `k8sPods` have anything to do with the kernel. Please add a comment describing what L156-L169 is doing

Reply:
They don't have anything to do with the kernel, but the command prefix and path have to be slightly different for the command to work correctly. Added some comments.